File: nodes/median/median.process.js
export default async ({inputs, settings, config}) => {

  const numbers = Array.isArray(inputs.numbers) ? inputs.numbers : [inputs.numbers];
  
  // Validate all inputs are numbers
  for (const n of numbers) {
    if (n === null || n === undefined || isNaN(Number(n))) {
      throw new Error("Input contains non-numeric values that cannot be processed");
    }
  }
  
  // Convert all values to numbers
  const validNumbers = numbers.map(n => Number(n));
  
  if (validNumbers.length === 0) {
    return { result: 0 };
  }
  
  // Sort the numbers
  const sorted = [...validNumbers].sort((a, b) => a - b);
  
  // Find the median
  const mid = Math.floor(sorted.length / 2);
  const result = sorted.length % 2 === 0
    ? (sorted[mid - 1] + sorted[mid]) / 2
    : sorted[mid];
  
  return { result };
}; 

File: nodes/median/median.process.py
async def process(inputs, settings, config, nodeConfig):
  numbers = inputs.get("numbers", [])
  if not isinstance(numbers, list):
    numbers = [numbers]
    
  # Validate all inputs are numbers
  for n in numbers:
    try:
      float(n)
    except (ValueError, TypeError):
      raise ValueError("Input contains non-numeric values that cannot be processed")
  
  # Convert all values to numbers
  valid_numbers = [float(n) for n in numbers]
  
  if not valid_numbers:
    return {"result": 0}
    
  # Sort the numbers
  sorted_numbers = sorted(valid_numbers)
  
  # Find the median
  n = len(sorted_numbers)
  mid = n // 2
  
  if n % 2 == 0:
    # Even number of elements
    result = (sorted_numbers[mid - 1] + sorted_numbers[mid]) / 2
  else:
    # Odd number of elements
    result = sorted_numbers[mid]
    
  return {"result": result} 